### PR TITLE
Fix / stac json files generation

### PR DESCRIPTION
**Pull request by zcernigoj**
- added npm script for checking validity of stac json files with [`stac-node-validator`](https://www.npmjs.com/package/stac-node-validator)
  - executed by running `npm run test-stac`
  - not part of the husky pre-commit hook yet because a lot of files are still invalid
- _[point 1 in internal issue]_ added the protocol (`https://`) to the `Endpoint` URL under `Resources` in yaml files (gets transformed to `url` in elements in `providers` in stac json files)
- _[point 3 in internal issue]_ updated the PROJJSON schema from v0.2 to v0.4
- _[point 5 in internal issue]_ replaced `eo:gsd` with `gsd`
- _[point 6 in internal issue]_ changed `stac.hbs` so it adds `title` and `type` to the License `link` object in stac json files
  - before: `{ "href": "url...", "rel": "license" }`
  - after: `{ "href": "url...", "rel": "license", "title": "License", "type": "text/html" }`
  - similar as  https://github.com/eurodatacube/public-collections/pull/152 for `eo:bands` and `raster:bands`

[internal issue](https://git.sinergise.com/team-6/openeo-platform/-/issues/47)

### CODE REVIEW

File: _build/stac.hbs [hunk 1]
```diff
@@ -48,7 +48,9 @@
 {{/each}}
     {
       "href": "{{{escapeJson LicenseUrl}}}",
-      "rel": "license"
+      "rel": "license",
+      "title": "License",
+      "type": "text/html"
```
Comment (dgostencnik) [inline _build/stac.hbs:53]: Should this be mimeType or licence type? In yaml files sometimes there is `LicenseType` key with value `proprietary` which makes me believe this shouldn't be mime type

File: package.json [hunk 2]
```diff
@@ -7,7 +7,8 @@
     "postinstall": "husky install",
     "build": "./node_modules/gulp-cli/bin/gulp.js",
     "serve": "./node_modules/gulp-cli/bin/gulp.js serve",
-    "test": "jest"
+    "test": "jest",
+    "test-stac": "npx stac-node-validator ./_output/stac/*.json"
```
[the remark this answers is not in the document]
Comment (zcernigoj) [reply package.json:11]: I tried to make the same workflow as for `test` - which doesn't make a build before running the tests.

I agree with this but then I would also change the `test` script to do the same. cc @dthiex

File: _build/stac.hbs (hunk 1 above)
Comment (zcernigoj) [reply _build/stac.hbs:53]: This gets transformed to an object inside of the `links` array in stac json files where `type` tells the consumer (another script or a human) what mime type / format the link at that URL is. So, I think it should be mime type.

File: package.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (zcernigoj) [reply package.json]: Done.

**Review by dgostencnik — DISMISSED** [review]

**Comment by zcernigoj** [comment]
> Despite all the changes, half of generated json files are still invalid. Should those errors be also fixed?

Some of them will probably be fixed when https://github.com/eurodatacube/public-collections/pull/152 is merged. It would probably be best to fix the rest in a separate PR.

**Comment by zcernigoj** [comment]
I changed the `RegistryEntryLastModified` so the approval is again needed.

**Review by dgostencnik — APPROVED** [review]

**Pull request merged**